### PR TITLE
switch to using the official router helm chart

### PR DESCRIPTION
**Pull request by lennyburdette**
the only thing i can't replicate right now is the config checksum for config-only deploys

### CODE REVIEW

**Review by lleadbet — APPROVED** [review]
Changes look good, although semi-concerned about the thread in the ref arch channel; if it's something else (or transient), feel free to merge.

File: deploy/router/environments/prod.yaml [hunk 1]
```diff
@@ -1,6 +1,27 @@
-monitoring:
-  otel: true
-resources:
-  requests:
-    cpu: 500m
-    memory: 512Mi
+router:
+  router:
```
Comment (garypen) [inline deploy/router/environments/prod.yaml:2]: :) 
NIT: Maybe change your router chart dependency name to `router-chart` ?

File: deploy/router/templates/backendconfig.yaml [hunk 2]
```diff
@@ -1,12 +1,12 @@
...
 metadata:
   name: http-hc-config
 spec:
   healthCheck:
     checkIntervalSec: 15
-    port: {{ .Values.healthcheck.port }}
+    port: {{ splitList ":" ((index .Values.router.router.configuration "health-check").listen | default ":8088") | last }}
     type: HTTP
     requestPath: /health
 {{- end }}
```
Comment (garypen) [inline deploy/router/templates/backendconfig.yaml:12]: I *think*, but haven't tried it, that with the next release of the router chart, you'll be able to do this in the router chart.

File: deploy/router/templates/secretsprovider.yaml [hunk 3]
```diff
@@ -12,5 +12,5 @@ spec:
           key: pwd
   parameters:
     secrets: |
-      - resourceName: {{ .Values.apolloKeySecretResourceName }}
+      - resourceName: pwd
```
Comment (garypen) [inline deploy/router/templates/secretsprovider.yaml:15]: You'll be able to do secret management in the next release of the router chart.

**Review by garypen — APPROVED** [review]
Very glad that you can use our chart to get things done. I hope it will be even more useful in the next and subsequent releases.

**Comment by lennyburdette** [comment]
Got the ingress working!

**Pull request merged**